src/ima-engine-decorator.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+import {BaseEngineDecorator, FakeEvent, Error} from '@playkit-js/playkit-js';
+import {Ima} from './ima';
+
+/**
+ * Engine decorator for ima dai plugin.
+ * @class ImaDAIEngineDecorator
+ * @param {IEngine} engine - The HTML5 engine.
+ * @param {Ima} plugin - The ima dai plugin.
+ */
+class ImaEngineDecorator extends BaseEngineDecorator {
+  _src: String;
+  _plugin: Ima;
+
+  constructor(engine: IEngine, plugin: Ima) {
+    super(engine);
+    this._plugin = plugin;
+  }
+
+  dispatchEvent(event: FakeEvent): ?boolean {
+    if (this._plugin.isAdsPlayingCustomPlayback() && !!event.payload && event.payload.code === Error.Code.VIDEO_ERROR) {
+      this._src = this._engine.getVideoElement().src;
+      setTimeout(() => {
+        if (this._src === this._engine.getVideoElement().src) {
+          return super.dispatchEvent(event);
+        }
+      }, 0);
```

Comment:
return a boolean here (`event.defaultPrevented` see [here](https://github.com/kaltura/playkit-js/blob/272a10f08976f850d35038c46d36e5f709008522/src/event/fake-event-target.js#L102)) and change the return value to `boolean` (remove ?)